src/html.rs

```diff
@@ -95,88 +93,117 @@ pub fn walk_and_embed_assets(
-                                            &css_data,
-                                            true,
                                             &href_full_url,
-                                            opt_no_images,
+                                            false,
+                                            "text/css",
                                             opt_silent,
-                                        ),
-
-                                        // If a network error occured, warn
-                                        Err(e) => {
-                                            eprintln!("Warning: {}", e);
-
-                                            // If failed to resolve, replace with absolute URL
-                                            href_full_url
-                                        }
-                                    };
-
-                                    attr.value.clear();
-                                    attr.value.push_slice(&replacement_text);
+                                        ) {
+                                            // On successful retrieval, traverse CSS
+                                            Ok((css_data, _)) => resolve_css_imports(
+                                                cache,
+                                                client,
+                                                &css_data,
+                                                true,
+                                                &href_full_url,
+                                                opt_no_images,
+                                                opt_silent,
+                                            ),
+
+                                            // If a network error occured, warn
+                                            Err(e) => {
+                                                eprintln!("Warning: {}", e);
+
+                                                // If failed to resolve, replace with absolute URL
+                                                href_full_url
+                                            }
+                                        };
+
+                                        attr.value.clear();
+                                        attr.value.push_slice(&replacement_text);
+                                    }
                                 }
                             }
                         }
-                    } else {
-                        for attr in attrs_mut.iter_mut() {
-                            if &attr.name.local == "href" {
-                                let href_full_url =
-                                    resolve_url(&url, attr.value.as_ref()).unwrap_or_default();
+                        LinkType::Preload | LinkType::DnsPrefetch => {
+                            // Since all resources are embedded as data URL, preloading and prefetching are unnecessary.
```

Comment:
This is the last nit, I promise: could you please remove the dot at the end of the comment? :see_no_evil:

Reply:
Ah, I did not notice that. I'll remove it. Thank you for the catch